src/sources/http.rs

```diff
@@ -48,13 +51,16 @@ pub enum Encoding {
 
 impl HttpSource for SimpleHttpSource {
     fn build_event(&self, body: Bytes, header_map: HeaderMap) -> Result<Vec<Event>, ErrorMessage> {
-        decode_body(body, self.encoding)
+        decode_body(body, self.encoding, self.metrics_plus_mode)
             .map(|events| add_headers(events, &self.headers, header_map))
             .map(|mut events| {
                 // Add source type
                 let key = log_schema().source_type_key();
                 for event in events.iter_mut() {
-                    event.as_mut_log().try_insert(key, Bytes::from("http"));
+                    match event {
+                        Event::Log(event) => event.try_insert(key, Bytes::from("http")),
+                        Event::Metric(_) => (),
```

Comment:
I think we should attempt to add the source type as a tag on the metric to match how we add it as a field to a log event.